docs/src/main/sphinx/admin/fault-tolerant-execution.rst

```diff
@@ -0,0 +1,338 @@
+     - Description
+     - Default value
+     - Retry policy
+   * - ``fault-tolerant-execution-task-descriptor-storage-max-memory``
+     - Maximum amount of memory to be used to store task descriptors for fault
+       tolerant queries on coordinator. Extra memory is needed to be able to
+       reschedule tasks in case of a failure.
+     - (JVM heap size * 0.15)
+     - Only ``TASK``
+
+.. _fte-exchange-manager:
+
+Exchange manager
+----------------
+
+Exchange spooling is responsible for storaging and managing spooled data for
+fault-tolerant execution. By default, spooled data is stored in the coordinator
+node's memory and is subject to memory resource limits. You can configure a
+filesystem-based exchange manager that stores spooled data in a specified
+location, either a local filesystem or an S3-compatible storage server.
+
+To configure an exchange manager, create a new ``exchange-manager.properties``
+configuration file on the coordinator, such as
+``etc/exchange-manager.properties``. In this file, set the
+``exchange-manager.name`` configuration propertry to ``filesystem``, and
+additional configuration properties as needed for your storage solution.
+
+.. list-table:: Exchange manager configuration properties
+   :widths: 30, 40, 30
+   :header-rows: 1
+
+   * - Property name
+     - Description
+     - Default value
+   * - ``exchange.base-directory``
+     - The base directory URI location that the exchange manager uses to store
+       spooling data.
+     -
+   * - ``exchange.encryption-enabled``
+     - Enable encrypting of spooling data.
+     - ``false``
+   * - ``exchange.sink-buffer-pool-min-size``
+     - The minimum buffer pool size for an exchange sink. The larger the buffer
+       pool size, the larger the write parallelism and memory usage.
+     - ``10``
+   * - ``exchange.source-concurrent-reader``
+     - The number of concurrent readers to read from spooling storage. The
+       larger the number of concurrent readers, the larger the read parallelism
+       and memory usage.
+     - ``4``
+   * - ``exchange.s3.aws-access-key``
+     - AWS access key to use.
+     -
+   * - ``exchange.s3.aws-secret-key``
+     - AWS secret key to use.
+     -
+   * - ``exchange.s3.region``
+     - Region of the S3 bucket.
+     -
+   * - ``exchange.s3.endpoint``
+     - S3 storage endpoint server if using an S3-compatible storage system that
+       is not AWS. If using AWS S3, can be ignored.
+     -
+   * - ``exchange.s3.max-error-retries``
+     - Maximum number of times the exchange manager's S3 client should retry
+       a request.
+     - ``3``
+   * - ``exchange.s3.upload.part-size``
+     - Part size for S3 multi-part upload.
+     - ``5MB``
+
+The following example ``exchange-manager.properties`` configuration specifies a
+local directory, ``/tmp/trino-exchange-manager``, as the spooling storage
+destination. Note that a local directory can only be used in a standalone
+environment, not in a distributed cluster.
```

Comment:
For testing purposes it should also work with distributed cluster if directory is shared (e.g via NFS of SMB) and accessible from all worker nodes.

Reply:
I added this as an explicit note bubble for now, but do we want to actually dissuade using a local filesystem for exchange, or is it just not useful for production clusters?

Reply:
Probably it is not useful for production clusters. It was not built with prod clusters in mind but for demos and testing.